solutions/devicesimulation/armtemplate/template.json

```diff
@@ -0,0 +1,575 @@
+        },
+        "remoteEndpointCertificate": {
+            "type": "securestring",
+            "metadata": {
+                "description": "The certificate that needs to be uploaded to the VM"
+            }
+        },
+        "remoteEndpointCertificateKey": {
+            "type": "securestring",
+            "metadata": {
+                "description": "The certificate key that needs to be uploaded to the VM"
+            }
+        }
+    },
+    "variables": {
+        "location": "[resourceGroup().location]",
+        "documentDBApiVersion": "2016-03-19",
+        "documentDBResourceId": "[resourceId('Microsoft.DocumentDb/databaseAccounts', parameters('documentDBName'))]",
+        "iotHubApiVersion": "2017-01-19",
+        "iotHubResourceId": "[resourceId('Microsoft.Devices/Iothubs', parameters('iotHubName'))]",
+        "iotHubKeyName": "iothubowner",
+        "iotHubKeyResource": "[resourceId('Microsoft.Devices/Iothubs/Iothubkeys', parameters('iotHubName'), variables('iotHubKeyName'))]",
+        "storageApiVersion": "2017-06-01",
+        "storageResourceId": "[resourceId('Microsoft.Storage/storageAccounts/',  parameters('storageName'))]",
+        "computeApiVersion": "2017-03-30",
+        "vmFQDN": "[concat(parameters('vmName'), '.', variables('location'), '.', parameters('vmFQDNSuffix'))]",
+        "vmImagePublisher": "Canonical",
+        "vmImageOffer": "UbuntuServer",
+        "vmStorageAccountContainerName": "vhds",
+        "osDiskName": "[concat(parameters('solutionName'), '-osdisk')]",
+        "networkApiVersion": "2017-06-01",
+        "vnetName": "[concat(parameters('solutionName'), '-vnet')]",
+        "vnetResourceId": "[resourceId('Microsoft.Network/virtualNetworks',variables('vnetName'))]",
+        "vnetAddressPrefix": "10.0.0.0/16",
+        "networkSecurityGroupName": "[concat(parameters('solutionName'), '-nsg')]",
+        "subnetName": "default",
+        "subnetPrefix": "10.0.0.0/24",
+        "subnetResourceId": "[concat(variables('vnetResourceId'),'/subnets/',variables('subnetName'))]",
+        "publicIPName": "[concat(parameters('solutionName'), '-public-ip')]",
+        "nicName": "[concat(parameters('solutionName'), '-nic')]",
+        "appServiceSku": "S1",
+        "appServiceCapacity": "1",
+        "webAppRepoURL": "https://github.com/Azure/reverse-proxy-dotnet.git",
+        "webAppRepoBranch": "master",
+        "webAppName": "[parameters('azureWebsiteName')]"
+    },
+    "resources": [
+        {
+            "comments": "Azure IoT Hub",
+            "apiVersion": "[variables('iotHubApiVersion')]",
+            "type": "Microsoft.Devices/Iothubs",
+            "name": "[parameters('iotHubName')]",
+            "location": "[variables('location')]",
+            "sku": {
+                "name": "[parameters('iotHubSku')]",
+                "tier": "[parameters('iotHubTier')]",
+                "capacity": 1
+            },
+            "properties": {
```

Comment:
Besides `sku`, there is one more configurable property of IoTHub - partition count. It will affect the throughput by limit the most concurrent event processor. Shall we expose this property to end-user, just like we exposed SKU?

Reply:
we'll work on performance later, I think we can continue with 4 partitions for the time being